bvm/ballerina-core/src/main/java/org/ballerinalang/model/JSONDataSource.java

```diff
@@ -37,4 +38,9 @@
      */
     void serialize(JsonGenerator gen) throws IOException;
 
+    boolean hasNext();
```

Comment:
Shall we add documentation this, because it is a public API.

Reply:
Updated